core-api/src/main/java/com/optimizely/ab/config/audience/MatchType.java

```diff
@@ -0,0 +1,187 @@
+package com.optimizely.ab.config.audience;
```

Comment:
Please add unit tests for the Match types as well as the different Conditions classes, especially around the `evaluate` logic.